tests/test_backend.py

```diff
@@ -499,6 +504,41 @@ def test_gather(self):
         _ = tf.identity(x_, name=_TFOUTPUT)
         self._run_test_case([_OUTPUT], {_INPUT: x_val})
 
+    @unittest.skipIf(*support_op_with_target('rs6', 'GatherNd'))
+    def test_gathernd(self):
+        x_val = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]], dtype=np.float32)
+        indices = np.array([[[0, 1], [1, 1]], [[1, 2], [0, 2]]], dtype=np.int32)
+        x = tf.placeholder(tf.float32, x_val.shape, name=_TFINPUT)
+        x_ = tf.gather_nd(x, tf.constant(indices))
+        _ = tf.identity(x_, name=_TFOUTPUT)
+        self._run_test_case([_OUTPUT], {_INPUT: x_val})
+        tf.reset_default_graph()
+
+        x_val = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9], dtype=np.float32)
+        indices = np.array([[0], [2], [4], [7]], dtype=np.int32)
+        x = tf.placeholder(tf.float32, x_val.shape, name=_TFINPUT)
+        x_ = tf.gather_nd(x, tf.constant(indices))
+        _ = tf.identity(x_, name=_TFOUTPUT)
+        self._run_test_case([_OUTPUT], {_INPUT: x_val})
+
+    @unittest.skipIf(*support_op_with_target('rs6', 'GatherNd'))
+    def test_gathernd_less_index(self):
```

Comment:
it means index shape is less than params rank, so it will gather a vector rather than a scalar. for instance, in the first test, params (x_val) rank is 2 while index shape ([0]) is 1, it will gather [1,2,3] resemble with Gather.